File: Algorithms/Maths/factorial/factorial.py
def factorial(num):
# recursive function
    if num == 1:
        return num
    else:
        return num * factorial(num - 1)


num = int(input("Enter a Number: "))
if num < 0:
    print("Factorial cannot be found for negative numbers")
elif num == 0:
    print("Factorial of 0 is 1")
else:
    print("Factorial of", num, "is: ", factorial(num))

File: Algorithms/Maths/Factorial/Java/Factorial.java
public class Factorial {
  
  static int factorialIterative(int num) {
    if (num <= 1) return 1;

    int total = 1;

    for (int i = 1; i < num + 1; i++) {
      total *= i;
    }

    return total;
  }

  static int factorialRecursive(int num) {
    if (num <= 1) return 1;

    return num * factorialRecursive(num - 1);
  }

  public static void main(String[] args) {
    System.out.println(factorialIterative(5));
    System.out.println(factorialRecursive(5));
  }

}
